GPSTest/src/main/java/com/android/gpstest/GpsStatusFragment.java

```diff
@@ -350,14 +350,27 @@ public void onLocationChanged(Location location) {
         // Make sure TTFF is shown, if the TTFF is acquired before the mTTFFView is initialized
         mTTFFView.setText(mTtff);
 
-        boolean showDMS = Application.getPrefs().getBoolean(getString(R.string.pref_key_dms_mode), false);
-        if (showDMS) {
-            mLatitudeView.setText(UIUtils.getDMSFromLocation(Application.get(), location.getLatitude()));
-            mLongitudeView.setText(UIUtils.getDMSFromLocation(Application.get(), location.getLongitude()));
-        } else {
-            mLatitudeView.setText(mRes.getString(R.string.gps_latitude_value, location.getLatitude()));
-            mLongitudeView.setText(mRes.getString(R.string.gps_longitude_value, location.getLongitude()));
+        String gpsDisplayMode = Application.getPrefs().getString(getString(R.string.pref_key_preferred_display_mode), "dd");
+
+        switch ( gpsDisplayMode ) {
+            case "dd":
```

Comment:
Same here - all these strings for the cases should be defined in `do_not_translate.xml`.  You'll have to fetch all of them using `Application.getPrefs().getString()` prior to the switch statement.